src/Databases/SQLite/DatabaseSQLite.cpp

```diff
@@ -57,6 +59,9 @@ DatabaseTablesIteratorPtr DatabaseSQLite::getTablesIterator(ContextPtr local_con
 
 std::unordered_set<std::string> DatabaseSQLite::fetchTablesList() const
 {
+    if (!sqlite_db)
```

Comment:
Why it can be not opened if we call `openSQLiteDB` in constructor?

Reply:
Because database_path might not exist when database engine is attached. In this case exception might be thrown, but it is not allowed because the server will fail to startup. So if it does not exist, we ignore this error in database engine constructor and throw it later when database is accessed.

Reply:
And file can occur on any method call (`checkSQLiteTable`, `fetchTable`, `StorageSQLite::write`), so we have to check `sqlite_db` in all of them and initialize it if needed. Is it right?

Reply:
Yes

Reply:
Also, why it might not exist when database engine is attached? Should it exist on table creation?

Reply:
> Should it exist on table creation?

Yes, this is why there is a check via `args.attach` - if it is attach or not.

>  why it might not exist when database engine is attached

For mysterious reasons. (Or for example in stress test in ci)